lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -239,6 +240,15 @@ def add_customer_data(post, options)
         end
       end
 
+      # created a separate method for these fields because they should not be included
+      # in all transaction types that include methods with source and customer fields
+      def add_extra_customer_data(post, payment_method, options)
+        post[:source][:phone] = {}
+        post[:source][:phone][:number] = options[:phone] || options.dig(:billing_address, :phone)
```

Comment:
There's part of me that thinks we should also check for `billing_address.phone_number` but in reality, we probably just need to fix the root of that problem, not the symptom 😞

Reply:
I remember you wrangling with this but I think in my tired-ness I was thinking that `billing_address.phone_number` was dependent on the gateway but couldn't remember. Was working down to the wire but would love to talk about this more.